sdk/main/src/ECDSAPrivateKey.cc

```diff
@@ -149,7 +149,14 @@ std::vector<unsigned char> ECDSAPrivateKey::sign(const std::vector<unsigned char
     throw std::runtime_error(internal::OpenSSLHasher::getOpenSSLErrorMessage("EVP_MD_CTX_new"));
   }
 
-  if (EVP_DigestSignInit(messageDigestContext, nullptr, EVP_sha256(), nullptr, mKeypair) <= 0)
+  OSSL_LIB_CTX* libraryContext = OSSL_LIB_CTX_new();
```

Comment:
Any changes you decide the make to `PublicKey` should happen here too

Reply:
Added